src/nvim/statusline.c

```diff
@@ -1325,6 +1325,7 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, char *opt_n
       stl_items[curitem].type = ClickFunc;
       stl_items[curitem].start = out_p;
       stl_items[curitem].cmd = xmemdupz(t, (size_t)(fmt_p - t));
+      xfree(stl_items[curitem].cmd);
```

Comment:
It's just immediately freeing the memory allocated on the line above. Those two lines could as well be `stl_items[curitem].cmd = NULL;`, except now you additionally have a dangling pointer here.

Reply:
this was just temporary, but originally I saw the memcpy in the called function and assumed it was copied elsewhere(that does explain why the tests failed)

Reply:
https://github.com/neovim/neovim/blob/1484995a1e5ab29010878fa5ee27c935fec6522f/src/nvim/statusline.c#L2128

Here is the only time when `cmd` is used, it gives away the pointer to something else. You need to do basically the same thing in the `else` branch of `tabtab != NULL`, but free the memory instead:
```c
  } else {
    for (long l = 0; l < itemcnt; l++) {
      if (stl_items[l].type == ClickFunc) {
        XFREE_CLEAR(stl_items[l].cmd);
      }
    }
  }
```

Reply:
I think this would avoid the leak, there's no need to even alloc the click func I don't think:
```diff
       while (*fmt_p != STL_CLICK_FUNC && *fmt_p) {
         fmt_p++;
       }
-      if (*fmt_p != STL_CLICK_FUNC) {
+      if (*fmt_p != STL_CLICK_FUNC || tabtab == NULL) {
         break;
       }
       stl_items[curitem].type = ClickFunc;
```

Reply:
Yeah no this would be better, otherwise `nvim_eval_statusline()` would not return the string either:
```diff
@@ -1321,7 +1321,7 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, char *opt_n
       }
       stl_items[curitem].type = ClickFunc;
       stl_items[curitem].start = out_p;
-      stl_items[curitem].cmd = xmemdupz(t, (size_t)(fmt_p - t));
+      stl_items[curitem].cmd = tabtab ? xmemdupz(t, (size_t)(fmt_p - t)) : NULL;
       stl_items[curitem].minwid = minwid;
       fmt_p++;
       curitem++;
```